### PR TITLE
Update CONTRIBUTING.md to reflect "help wanted" label requirement

### PR DESCRIPTION
**Pull request by asfalcone**
PRs should have an accompanying "help wanted" issue that they are targeting

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,9 +74,9 @@ If you're happy with what you see, leave a comment on the GitHub issue stating y
 
 Pull requests are always welcome :simple_smile:
 
-If you're working on fixing a particular issue, referring to it in the description of your PR is really helpful for establishing the context needed for review.
+PRs should reference an open github issue marked with the [help wanted](https://github.com/fastlane/fastlane/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) label. Referring to the issue in the description of your PR is required and is really helpful for establishing the context needed for review.
 
-If your PR is contributing new functionality, did you create an issue to discuss it with the community first? Great! :raised_hands: Reference that issue in your PR.
+If you're contibuting new functionality, please open a new issue explaining the functionality desired first so that we can discuss as a community. We'll add the [help wanted](https://github.com/fastlane/fastlane/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) label if we believe this to be a meaningful contribution that will benefit other fastlane users and you go ahead with the pull request. :raised_hands:
```
Comment (mfurtak) [inline CONTRIBUTING.md]: What do you think about reversing the order of this paragraph and the previous one? This one seems to be step 1.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -82,9 +74,9 @@ If you're happy with what you see, leave a comment on the GitHub issue stating y
 
 Pull requests are always welcome :simple_smile:
 
-If you're working on fixing a particular issue, referring to it in the description of your PR is really helpful for establishing the context needed for review.
+PRs should reference an open github issue marked with the [help wanted](https://github.com/fastlane/fastlane/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) label. Referring to the issue in the description of your PR is required and is really helpful for establishing the context needed for review.
```
Comment (mfurtak) [inline CONTRIBUTING.md]: nitpick: github -> GitHub

File: CONTRIBUTING.md (hunk 1 above)
Comment (ohwutup) [inline CONTRIBUTING.md]: also maybe bolding or italicizing "first"?

**Comment by mfurtak** [comment]
1 suggestion and 1 nitpick, otherwise LGTM!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -93,6 +85,12 @@ If your PR is contributing new functionality, did you create an issue to discuss
     - You can run the tests by executing `bundle install` and then `bundle exec rspec`.
 - Your code editor should indent using spaces with a tab size of 2 spaces.
 
+To submit the changes to the fastlane repo, you have to do the following:
+
+- Squash your commits into one. For example, to squash three commits into one, do the following: `$ git rebase -i HEAD~3`. In the text editor that comes up, replace the words "pick" with "squash" next to the commits you want to squash. Save and close the editor. For more information, take a look at [7.6 Git Tools - Rewriting History](http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)
```
Comment (KrauseFx) [inline CONTRIBUTING.md]: We can now do that with the new GitHub feature, right? https://github.com/blog/2141-squash-your-commits

**Comment by KrauseFx** [comment]
Also added a comment on this

File: CONTRIBUTING.md (hunk 3 above)
Comment (asfalcone) [inline CONTRIBUTING.md]: I'll update the squashing instructions in a separate PR once we figure out what repo settings we're going to use

**Pull request merged**